pkg/dogstatsd/listeners/uds_common.go

```diff
@@ -149,15 +185,27 @@ func (l *UDSListener) Listen() {
 				packet.Origin = container
 			}
 			// Return the buffer back to the pool for reuse
-			l.oobPool.Put(oob)
+			l.oobPoolManager.Put(oob)
 		} else {
 			t2 = time.Now()
 			tlmListener.Observe(float64(t2.Sub(t1).Nanoseconds()), "uds")
 
 			// Read only datagram contents with no credentials
-			n, _, err = l.conn.ReadFromUnix(packet.buffer)
+			n, _, err = l.conn.ReadFromUnix(packet.Buffer)
 
 			t1 = time.Now()
+
+			if capBuff != nil {
+				capBuff.Pb.Timestamp = time.Now().Unix()
+				capBuff.Buff = packet
+				capBuff.Pb.AncillarySize = int32(0)
+				capBuff.Pb.PayloadSize = int32(n)
+				capBuff.Pb.Payload = packet.Buffer // or packet.Buffer[:n] ?
+			}
+		}
+
+		if capBuff != nil {
+			l.trafficCapture.Writer.Enqueue(capBuff)
```

Comment:
As `Enqueue` put `capBuff` in a channel, you have to make sure all references inside `capBuff` stays valid. 
After `l.oobPoolManager.Put(oob)` , the instance `oob` can be reused and overwritten. As `oob` is store as pointer (`capBuff.Oob = &oob`), you may have a race condition.

I did not check, but there may have a similar issue for `capBuff.Pb.Payload = packet.Buffer[:n]`